core/base/src/main/java/org/eclipse/dataspaceconnector/core/config/ConfigImpl.java

```diff
@@ -86,6 +85,23 @@ public Long getLong(String key, Long defaultValue) {
         return getNumber(key, defaultValue, "long", Long::parseLong);
     }
 
+    @Override
+    public Boolean getBoolean(String key) {
+        return getNotNullValue(key, this::getBoolean);
+    }
+
+    @Override
+    public Boolean getBoolean(String key, Boolean defaultValue) {
```

Comment:
Yes, I also see that, but since parseBoolean also parses any strings as `false` which are not `"true"` (ignoreCase), a settings typo like `"truee"` would also be parsed as `false` and I can imagine long debugging sessions to find the typo in the settings. Didn't know if we wanted it that way, if it doesn't matter, I can use the above `getNumber`.
`Result of 'Boolean.parseBoolean("truee")' is always 'false' `

That's why I intercepted it by:
`!value.equalsIgnoreCase("true") && !value.equalsIgnoreCase("false")` -> `EdcException`

What is the opinion here? Can change it otherwise.

Reply:
The `getNumber` (to be eventually renamed) accept a parse function that will parse `String` to `Boolean`, if the parser throws exception, it gets managed, so it's up to that parse function to handle this logic.
So that condition checking could be done inside inside the parse function, that could be implemented also as:
```
if (value.equalsIgnoreCase("true")) return true;
if (value.equalsIgnoreCase("false")) return false;
throw new EdcException("cannot parse");
```
the exception will be catched

Reply:
Done (please check again if ok)